src/common/utils/http.ts

```diff
@@ -0,0 +1,48 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { container } from "readium-desktop/main/di";
+import { RootState } from "readium-desktop/main/redux/states";
+import { Store } from "redux";
+import * as request from "request";
+import { promisify } from "util";
+
+export async function httpGet(url: string, options?: request.CoreOptions): Promise<string> {
```

Comment:
if `options.json` === `true` (or truthy), then `httpGet()` returns a promise of type `any` (raw JSON), not `string`. Therefore, I suggest the function signature explicitly returns `Promise<string | any>`, with a code comment to explain the reason for `any` in the TypeScript union type.

The alternative option would be to `delete options.json`, but I feel uncomfortable about this, because this breaks the API contract of the original function, and requires adapting the client-side consumer code (i.e. to add explicit `JSON.parse()`).

What do you think?

Reply:
i didn't know about that, I removed the returns type

Reply:
I see you removed the TypeScript return type in the function signature, why not use the union type `Promise<string | any>`? (yes, `string` is redundant with `any`, but there is an API comment to explain `json:true` => `any`).

Reply:
if you want, yes it's redundant

Reply:
![Screenshot 2019-07-29 at 16 05 57](https://user-images.githubusercontent.com/3198865/62054891-cd040f80-b21a-11e9-8085-564c3e6130be.png)

the interest is really limited anyway.

Reply:
Thanks. This way, both the TypeScript compiler and the human programmer have an explicit reference to the `json` edge case.

Reply:
yes and no the string return type option is only available when the dev see the file. The information panel, so also the typescript compiler,  overload string type with any.
Cf image above.

Yes the human programmer can have an explicit reference but more with the comment on function returns type

Reply:
Well, instead of `any` we can use the following `AnyJson` type:

`declarations.ts`:
```
export type AnyJson =  JsonPrimitives | JsonArray | JsonMap;
export type JsonPrimitives = string | number | boolean | null;
export interface JsonMap {  [key: string]: AnyJson; }
export interface JsonArray extends Array<AnyJson> {}
```

Reply:
Thanks i apply it

Reply:
Do you think we can use `typings.d.ts`? (that is what I meant by `declarations.d.ts`)
https://github.com/readium/readium-desktop/blob/develop/typings.d.ts

Reply:
we can't use the typing declaration with interface , no ?
I created a new file.ts especially for this

Reply:
We have to create a new module otherwise

Reply:
Ok, thanks :)